File: 1910-remove_all_occurrences_of_a_substring/go/main.go
package main

import (
	"fmt"
	"strings"
)

func removeOccurrences(s string, part string) string {
	for strings.Contains(s, part) {
		index := strings.Index(s, part)
		s = s[:index] + s[index+len(part):]
	}
	return s
}

func main() {
	s := "daabcbaabcbc"
	part := "abc"
	result := removeOccurrences(s, part)
	fmt.Printf("Result: %#v\n", result)
	if result == "dab" {
		fmt.Println("Passed")
	} else {
		fmt.Println("Failed")
	}
}

File: 1910-remove_all_occurrences_of_a_substring/ruby/main.rb
# @param {String} s
# @param {String} part
# @return {String}
def remove_occurrences(s, part)
  while s.include?(part)
    index = s.index(part)
    s = s[0, index] + s[index + part.length, s.length]
  end
  return s
end

if __FILE__ == $0
  s = "daabcbaabcbc"
  part = "abc"
  result = remove_occurrences(s, part)
  puts "Result: #{result}"

  if result == "dab"
    puts "Passed"
  else
    puts "Failed"
  end
end

